sdk/keyvault/azure-keyvault-keys/tests/_test_case.py

```diff
@@ -2,18 +2,61 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 # ------------------------------------
+import functools
 import os
 
 from azure.keyvault.keys._shared import HttpChallengeCache
+from azure.keyvault.keys._shared.client_base import ApiVersion, DEFAULT_VERSION
 from devtools_testutils import AzureTestCase
-from parameterized import parameterized
+from parameterized import parameterized, param
 import pytest
 from six.moves.urllib_parse import urlparse
 
 
+def client_setup(testcase_func):
+    """decorator that creates a client to be passed in to a test method"""
+    @functools.wraps(testcase_func)
+    def wrapper(test_class_instance, api_version, is_hsm=False, **kwargs):
+        test_class_instance._skip_if_not_configured(api_version, is_hsm)
+        endpoint_url = test_class_instance.managed_hsm_url if is_hsm else test_class_instance.vault_url
+        client = test_class_instance.create_key_client(endpoint_url, api_version=api_version, **kwargs)
+
+        if kwargs.get("is_async"):
+            import asyncio
+
+            coroutine = asyncio.coroutine(testcase_func)
+            future = coroutine(test_class_instance, client, is_hsm=is_hsm)
+            loop = asyncio.get_event_loop()
+            loop.run_until_complete(future)
```

Comment:
If `testcase_func` is an `async` function, it returns a `coroutine`:
```py
>>> async def foo(): pass;
>>> foo()
<coroutine object foo at 0x7f4ce239ac50>
```
So you don't need to create one:
```suggestion
            coroutine = testcase_func(test_class_instance, client, is_hsm=is_hsm)
            loop = asyncio.get_event_loop()
            loop.run_until_complete(coroutine)
```